java/ingest/src/main/java/sleeper/ingest/impl/StandardIngestCoordinator.java

```diff
@@ -86,130 +91,15 @@ public static IngestCoordinator<Record> asyncS3WriteBackedByArrow(IngestProperti
-
-        public BackedByArrowBuilder workingArrowBufferAllocatorBytes(long workingArrowBufferAllocatorBytes) {
-            this.workingArrowBufferAllocatorBytes = workingArrowBufferAllocatorBytes;
-            return this;
-        }
-
-        public BackedByArrowBuilder minBatchArrowBufferAllocatorBytes(long minBatchArrowBufferAllocatorBytes) {
-            this.minBatchArrowBufferAllocatorBytes = minBatchArrowBufferAllocatorBytes;
-            return this;
-        }
-
-        public BackedByArrowBuilder maxBatchArrowBufferAllocatorBytes(long maxBatchArrowBufferAllocatorBytes) {
-            this.maxBatchArrowBufferAllocatorBytes = maxBatchArrowBufferAllocatorBytes;
-            return this;
-        }
-
-        public BackedByArrowBuilder maxNoOfBytesToWriteLocally(long maxNoOfBytesToWriteLocally) {
-            this.maxNoOfBytesToWriteLocally = maxNoOfBytesToWriteLocally;
-            return this;
-        }
-
-        public IngestCoordinator<Record> buildDirectWrite(String filePathPrefix) {
-            return builder.buildIngestCoordinator(
-                    buildRecordBatchFactoryFn(),
-                    builder.buildDirectFileWriterFactory(filePathPrefix));
-        }
-
-        public IngestCoordinator<Record> buildAsyncS3Write(String s3BucketName, S3AsyncClient s3AsyncClient) {
-            return builder.buildIngestCoordinator(
-                    buildRecordBatchFactoryFn(),
-                    builder.buildAsyncFileWriterFactory(s3BucketName, s3AsyncClient));
-        }
-
-        private Supplier<RecordBatch<Record>> buildRecordBatchFactoryFn() {
-            return () ->
-                    new ArrowRecordBatchAcceptingRecords(
-                            arrowBufferAllocator,
-                            builder.schema,
-                            builder.localWorkingDirectory,
-                            workingArrowBufferAllocatorBytes,
-                            minBatchArrowBufferAllocatorBytes,
-                            maxBatchArrowBufferAllocatorBytes,
-                            maxNoOfBytesToWriteLocally,
-                            maxNoOfRecordsToWriteToArrowFileAtOnce);
-        }
-    }
-
-    public interface BackedBuilder {
-        IngestCoordinator<Record> buildDirectWrite(String filePathPrefix);
-
-        IngestCoordinator<Record> buildAsyncS3Write(String s3BucketName, S3AsyncClient s3AsyncClient);
+                .arrowRecordBatchFactory(builder -> builder
+                        .bufferAllocator(arrowBufferAllocator)
+                        .maxNoOfRecordsToWriteToArrowFileAtOnce(maxNoOfRecordsToWriteToArrowFileAtOnce)
+                        .workingBufferAllocatorBytes(workingArrowBufferAllocatorBytes)
+                        .minBatchBufferAllocatorBytes(minBatchArrowBufferAllocatorBytes)
+                        .maxBatchBufferAllocatorBytes(maxBatchArrowBufferAllocatorBytes)
+                        .maxNoOfBytesToWriteLocally(ingestProperties.getMaxRecordsToWriteLocally()))
```

Comment:
This is incorrect, the records to write locally and bytes to write locally are two separate things.